src/devices/qa.ts

```diff
@@ -0,0 +1,540 @@
+        extend: [
+            tuya.modernExtend.tuyaMagicPacket(),
+            deviceEndpoints({endpoints: {'l1': 1, 'l2': 2, 'l3': 3, 'l4': 4}}),
+            tuya.modernExtend.tuyaOnOff({endpoints: ['l1', 'l2', 'l3', 'l4'], powerOnBehavior2: true, switchType: true}),
+        ],
+    },
+    {
+        fingerprint: tuya.fingerprint('TS0001', [
+            '_TZ3000_dov0a3p1',
+        ]),
+        model: 'QAT42Z1H',
+        vendor: 'QA',
+        description: '1 channel wall switch',
+        extend: [
+            tuya.modernExtend.tuyaMagicPacket(),
+            deviceEndpoints({endpoints: {'l1': 1}}),
+            tuya.modernExtend.tuyaOnOff({endpoints: ['l1'], powerOnBehavior2: true, backlightModeOffOn: true}),
+        ],
+    },
+    {
+        fingerprint: tuya.fingerprint('TS0002', [
+            '_TZ3000_gkesadus',
+        ]),
+        model: 'QAT42Z2H',
+        vendor: 'QA',
+        description: '2 channel wall switch',
+        extend: [
+            tuya.modernExtend.tuyaMagicPacket(),
+            deviceEndpoints({endpoints: {'l1': 1, 'l2': 2}}),
+            tuya.modernExtend.tuyaOnOff({endpoints: ['l1', 'l2'], powerOnBehavior2: true, backlightModeOffOn: true}),
+        ],
+    },
+    {
+        fingerprint: tuya.fingerprint('TS0003', [
+            '_TZ3000_pmsxmttq',
+        ]),
+        model: 'QAT42Z3H',
+        vendor: 'QA',
+        description: '3 channel wall switch',
+        extend: [
+            tuya.modernExtend.tuyaMagicPacket(),
+            deviceEndpoints({endpoints: {'left': 1, 'center': 2, 'right': 3}}),
+            tuya.modernExtend.tuyaOnOff({endpoints: ['left', 'center', 'right'], powerOnBehavior2: true, backlightModeOffOn: true}),
+        ],
+    },
+    {
+        fingerprint: tuya.fingerprint('TS0601', [
+            '_TZE204_4cl0dzt4',
+        ]),
+        model: 'QAT44Z6H',
+        vendor: 'QA',
+        description: '6 channel wall switch',
+        exposes: [e.switch().withEndpoint('l1').setAccess('state', ea.STATE_SET),
+            e.switch().withEndpoint('l2').setAccess('state', ea.STATE_SET),
+            e.switch().withEndpoint('l3').setAccess('state', ea.STATE_SET),
+            e.switch().withEndpoint('l4').setAccess('state', ea.STATE_SET),
+            e.switch().withEndpoint('l5').setAccess('state', ea.STATE_SET),
+            e.switch().withEndpoint('l6').setAccess('state', ea.STATE_SET)],
+        fromZigbee: [fz.ignore_basic_report, legacy.fz.tuya_switch],
+        toZigbee: [legacy.tz.tuya_switch_state],
+        meta: {multiEndpoint: true},
+        endpoint: (device) => {
+            return {'l1': 1, 'l2': 1, 'l3': 1, 'l4': 1, 'l5': 1, 'l6': 1};
+        },
+        configure: async (device, coordinatorEndpoint, logger) => {
+            await reporting.bind(device.getEndpoint(1), coordinatorEndpoint, ['genOnOff']);
+            if (device.getEndpoint(2)) await reporting.bind(device.getEndpoint(2), coordinatorEndpoint, ['genOnOff']);
```

Comment:
You can remove the `if (device.getEndpoint(2)`, I guess the device always has these endpoints.

Reply:
Done.